chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -143,23 +145,85 @@ impl TryFrom<&PeerInfo> for WhitelistNode {
     }
 }
 
+pub(crate) struct PeerManagerState {
+    /// PeerManager config.
+    pub config: Arc<NetworkConfig>,
+    /// GenesisId of the chain.
+    pub genesis_id: GenesisId,
+    /// Address of the client actor.
+    pub client_addr: Recipient<NetworkClientMessages>,
+    /// Address of the view client actor.
+    pub view_client_addr: Recipient<NetworkViewClientMessages>,
+
+    /// Network-related info about the chain.
+    pub chain_info: RwLock<ChainInfo>,
+    /// AccountsData for TIER1 accounts.
+    pub accounts_data: Arc<accounts_data::Cache>,
+    /// Connected peers (inbound and outbound) with their full peer information.
+    pub connected_peers: RwLock<HashMap<PeerId, ConnectedPeer>>,
+}
+
+impl PeerManagerState {
+    /// Broadcast message to all active peers.
+    pub async fn broadcast_message(self: Arc<Self>, msg: SendMessage) {
+        metrics::BROADCAST_MESSAGES.with_label_values(&[msg.message.msg_variant()]).inc();
+        // Snapshot list of peers.
+        let peer_addrs: Vec<_> =
+            self.connected_peers.read().values().map(|p| p.addr.clone()).collect();
+        // Change message to reference counted to allow sharing with all actors
+        // without cloning.
+        let msg = Arc::new(msg);
+        // Send the message to each peer synchronously.
+        // TODO(gprusak): should it be asynchronous instead? The previous semantics was unclear.
+        for addr in peer_addrs {
+            addr.send(msg.clone())
+                .await
+                .expect("Failed sending broadcast message(broadcast_message)");
```

Comment:
Hm, this feels like it very much can trigger? Peers can get banned or unregistered, which stops the actor, which will lead to closed mailbox error.

Reply:
right, good catch, fixed